### PR TITLE
feat(overlays): add global backdrop opacity variable for animations

### LINKED ISSUE #16446: [v4] Backdrop opacity
image

There's already `backdrop-color` variable for customization. but `opacity` is hard-coded in the `element.style`.

It would be great to give a change to modify backdrop opacity with global variable.

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: fixes #16446

## What is the new behavior?

- Adds `--backdrop-opacity` as a component level variable to all overlays
- Adds `--ion-backdrop-opacity` to change the overlays backdrop opacity globally
- This removes the need to customize the entire animation

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/action-sheet/test/basic/index.html [hunk 1]
```diff
@@ -28,6 +28,7 @@
       <ion-button expand="block" id="noBackdropDismiss" onclick="presentNoBackdropDismiss()">No Backdrop Dismiss</ion-button>
       <ion-button expand="block" id="scrollableOptions" onclick="presentScroll()">Scrollable Options</ion-button>
       <ion-button expand="block" id="scrollWithoutCancel" onclick="presentScrollNoCancel()">Scroll Without Cancel</ion-button>
+      <ion-button expand="block" id="customBackdrop" onclick="presentWithCssClass('custom-backdrop')">Custom Backdrop Opacity</ion-button>
```
Comment (liamdebeasi) [inline core/src/components/action-sheet/test/basic/index.html]: Only suggestion would be to add this to the E2E test as well. Looks like it's testing all the other action sheet variants there as well.

**Comment by ShaggyDude** [comment]
Does this work now? I'm in the latest ionic 5 and it doesn't appear to work there?

**Comment by lincolnthree** [comment]
Works here. 
In your root stylesheet:

````
:root {
  --ion-backdrop-opacity: 1;
}```

**Comment by lincolnthree** [comment]
What doesn't work, however, is backdrop opacity on the ion-menu component. It does not seem to be integrated there.

**Comment by ShaggyDude** [comment]
lincolnthree is correct. Is there a way to make it work in ion-menu?

**Comment by alexandre-khoury** [comment]
same question here

**Pull request merged**